FinancialTracker/Firebase/DatabaseManager.swift

```diff
@@ -14,6 +14,19 @@ enum DatabaseError: Error {
     case nonExistingUser
     case access(String?)
     case unknown
+    
+    var description: String {
+        switch self {
```

Comment:
when you `switch` on `self`, you can also infer the enum's associated value and use that in your error's description. 

Also, if you are adding a description to an enum, use the `CustomStringConvertible` protocol.

Reply:
```swift
 var description: String {
         switch self {
           .....
           case access(let errorDescription): 
                     return "Access Error with description: \(errorDescription)"
             }
```

Reply:
If you do the above, you won't need to the switch altogether in the functions `create(basedOn ...`. 
Ok, as I look at it, you want to do a title + message for the alert, in that case, create an additional computed property `message` for the `DatabaseError` and the `AuthError` instead of switching within the one more time in the `create` functions.

No you are just half way with using the most of your enums.